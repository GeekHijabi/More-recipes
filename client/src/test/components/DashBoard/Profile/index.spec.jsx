import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedProfile, { Profile }
  from '../../../../js/components/DashBoard/Profile';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const props = {
  apiUpdateUserProfile: jest.fn(() => Promise.resolve({ ok: true })),
  apiGetCurrentUser: jest.fn(() => Promise.resolve({ ok: true })),
  currentUser: {
    id: 1,
    userName: 'tester',
    bio: '',
    summary: ''
  }
};


describe('Profile snapshot', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Profile {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('toggle', () => {
  it('should toggle the modal', () => {
    const wrapper = shallow(<Profile {...props} />);
    wrapper.setState({ modal: true });
    wrapper.instance().toggle();
    expect(wrapper.state().modal).toEqual(false);
  });
});

describe('Connected Profile component', () => {
  it('component successfully rendered', () => {
    const store = mockStore({
      auth: {
        currentUser: {}
      }
    });
    const wrapper = shallow(<ConnectedProfile store={store} />);
    expect(wrapper.length).toBe(1);
  });
});
