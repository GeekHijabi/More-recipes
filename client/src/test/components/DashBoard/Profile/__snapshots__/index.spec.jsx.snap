// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapshot should render correctly 1`] = `
<div
  page="profile"
>
  <withRouter(Connect(RecipeHeader)) />
  <main
    className="row prof-up"
  >
    <div
      className="dual col-xs-12 col-sm-6 col-md-6 col-lg-5"
    >
      <div
        className="mt-3"
      >
        <span
          className="float-right span-profile"
          data-toggle="tooltip"
          onClick={[Function]}
          role="button"
          tabIndex="-1"
          title="Edit profile"
        >
          <i
            className="fa fa-edit fa-2x"
          />
        </span>
        <h1
          className="name mb-5"
        >
          tester
        </h1>
        <h4>
          Summary
        </h4>
        <p
          className="name-desc"
        />
        <div
          className="red-border-line mb-4"
        />
        <h4>
          About Me
        </h4>
        <p />
        <Connect(EditProfileModal)
          editProfile={[MockFunction]}
          isOpen={false}
          toggle={[Function]}
          user={
            Object {
              "bio": "",
              "id": 1,
              "summary": "",
              "userName": "tester",
            }
          }
        />
      </div>
      <div
        className="social-icons"
      >
        <a
          href="/"
        >
          <i
            className="fa fa-twitter-square fa-2x"
          />
        </a>
        <a
          href="/"
        >
          <i
            className="fa fa-instagram fa-2x"
          />
        </a>
        <a
          href="/"
        >
          <i
            className="fa fa-pinterest-square fa-2x"
          />
        </a>
      </div>
    </div>
    <div
      className="dual col-xs-12 col-sm-6 col-md-6 col-lg-7"
    >
      <img
        alt=""
        className="bg"
        src={Object {}}
      />
    </div>
  </main>
  <Footer />
</div>
`;
